db/collection.go

```diff
@@ -56,6 +57,10 @@ type collection struct {
 	schemaID string
 
 	desc client.CollectionDescription
+
+	isIndexCached  bool
```

Comment:
todo: This bool represents a race condition, where it can be set to true, and then meanwhile, whilst collection is in use, an index is mutated/added/etc.

I am not sure we can be sure that this will only ever cause performance issues, especially regarding unique indexes and backfill.

I would suggest a rethink here.  Please bear in mind that the collection may currently outlive any given transaction that may be in scope at any given point in time (I dislike this anyway, but it is how the current system behaves, and we either need to change that, or handle it for stuff like this).